jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -53,203 +49,228 @@ def __init__(self, input_models, reffiles=None, **pars):
-                nc = len(this_channel)
-                for k in range(nc):
-                    self.band_name.append('ch' + this_channel[k] + '_' + this_subchannel)
-            elif self.instrument == 'NIRSPEC':
-                self.gratings.append(self.input_models[i].meta.instrument.grating.lower())
-            else:
-                # add error
-                raise ErrorWrongInstrument('Instrument must be MIRI or NIRSPEC')
-        if self.instrument == 'MIRI':
-            band_no_repeat = list(set(self.band_name))
-            bands = ['short','medium','long','short-medium','short-long','medium-short','medium-long','long-short','long-medium']
-            channels = ['1','2','3','4']
-            for this_band in band_no_repeat:
-                for j in channels:
-                    check_channel = 'ch' + j
-                    if check_channel in this_band:
-                        self.channels.append(j)
-                for k in bands:
-                    compare_band = this_band[4:]  # remove the channel part of the name to see which band we have
-                    if k == compare_band:
-                        self.subchannels.append(k)
-            self.ifu_band1 = self.channels
-            self.ifu_band2 = self.subchannels
-        elif self.instrument == 'NIRSPEC':
-            self.gratings = list(set(self.gratings))
-            self.ifu_band1 = self.gratings
-            self.ifu_band2 = self.gratings  # not used in NIRSpec
-
-    def _convert_inputs(self):
+        OutlierDetection.__init__(self, input_models, reffiles=reffiles, **pars)
+
+    def create_optional_results_model(self, opt_info):
+        """
+        Creates an OutlierOutputModel from the computed arrays from outlier detection on IFU data.
+
+        Parameter
+        ---------
+        input_model: ~jwst.datamodels.RampModel
+
+        opt_info: tuple
+        The output arrays needed for the OultierOutputModel.
+
+        Return
+        ---------
+        opt_model: OultierOutputModel
+        The optional OutlierOutputModel to be returned from the outlier_detection_ifu step.
+        """
+        (kernsize_x, kernsize_y, threshold_percent,
+         diffarr, minarr, normarr, minnorm) = opt_info
+        opt_model = datamodels.OutlierIFUOutputModel(
+            diffarr=diffarr,
+            minarr=minarr,
+            normarr=normarr,
+            minnorm=minnorm)
+        opt_model.meta.kernel_xsize = kernsize_x
+        opt_model.meta.kernel_ysize = kernsize_y
+        opt_model.meta.threshold_percent = threshold_percent
+        return opt_model
+
+    def _find_detector_parameters(self):
+        if self.input_models[0].meta.instrument.name.upper() == 'MIRI':
+            diffaxis = 1
+        elif self.input_models[0].meta.instrument.name.upper() == 'NIRSPEC':
+            diffaxis = 0
+        ny, nx = self.inputs[0].data.shape
+        return (diffaxis, ny, nx)
+
+    def convert_inputs(self):
```

Comment:
Why is this method necessary? I don't see the `converted` flag used anywhere. Is it to comply with the base class?